### ISSUE TITLE
[Shapes] Collection of highlight/selection visual issues

### ISSUE THREAD

**Issue by psobolewskiPhD** [napari/napari#7244]
## 🧰 Task

This is a meta issue for various Shapes selection & highlight issues:

- [x] With a Shape selected, highlight-on-hover is glitchy: sometimes doesn’t fire, sometimes (in Vertex selection mode) removes the selection highlight leaving just vertexes
- With a Shape selected, Shift-Click doesn’t visually select a 2nd Shape, even though it will be selected — unlike the drag selection (which also has buggy highlight of the shapes)
For these two see:

https://github.com/user-attachments/assets/b9f6bb38-ac04-4f71-a1f4-f9594a26b67f

Note: both of the above work correctly in 0.4.19post1, but not on main (with exception of the difference between shift-click multi selection and drag multi selection)

- https://github.com/napari/napari/issues/7073 clicking on a Shape in Pan_Zoom mode highlights Shape

Minor:
- https://github.com/napari/napari/issues/7074 shape under mouse is highlighted in Pan-zoom mode or drawing modes when application loses focus.

**Comment by psobolewskiPhD** [comment]
If I have a mix of rectangle/ellipse and the polygon lasso shapes, then if you have the polygon selected and mouse-over the other shapes a highlight edge is drawn to the other shape. 
![image](https://github.com/user-attachments/assets/14ce996d-bf96-4d07-95a2-8a5bbad35d13)
If you have the other shape selected and mouse-over the polygon, then a highlight edge is drawn to the polygon.

**Comment by psobolewskiPhD** [comment]
Post https://github.com/napari/napari/pull/7268 
I now see some "filled" triangles when mousing around in addition to the line glitch.
<img width="448" alt="image" src="https://github.com/user-attachments/assets/35284d67-d253-44eb-bdcc-0b1f8a092698">

**Comment by jni** [comment]
Ok I can reproduce the line to the next poly. It might have something to do with removing duplicates in one context but not the other. I'm not sure it's a release blocker though. 🤔 (we could make another release next week, too.)

**Comment by jni** [comment]
I can't reproduce the hell from [this Zulip message though 😂](https://napari.zulipchat.com/#narrow/channel/212875-general/topic/0.2E5.2E5/near/488004095). Which is also a good sign!

**Comment by jni** [comment]
I can confirm that the weird connecting lines are in 0.5.4 also. So this is an old issue and shouldn't block 0.5.5.

**Comment by jni** [comment]
btw, both on 0.5.4 and main, the lasso seems (to me) to be overzealous in deleting recent vertices — is that expected? CC @melonora 

https://github.com/user-attachments/assets/efd8c57c-46d8-4a81-b306-c4164d6a6f16

**Comment by melonora** [comment]
Hmm how high do you have the rdp set?

**Comment by psobolewskiPhD** [comment]
@jni with main and two polygons near each other I consistently get:
<img width="1337" alt="image" src="https://github.com/user-attachments/assets/24f350ad-9896-463d-96f0-f7e5d0ec9222" />

**Comment by psobolewskiPhD** [comment]
The lasso thing feels like something we discussed in a different PR: should the last click/double-click count as a vertex or no. But i've not delved.

**Comment by jni** [comment]
> @jni with main and two polygons near each other I consistently get: 

It would be super handy if you could tell me if you see these issues in 0.5.4 also. 🙏

**Comment by jni** [comment]
Also, can you please save your coordinates when you get such a case so we can reproduce? Here's my admittedly crude attempt:

<img width="1179" alt="Screenshot 2024-12-12 at 8 59 17 pm" src="https://github.com/user-attachments/assets/79a0cfa4-89e3-4126-b742-82c30f527ede" />

**Comment by psobolewskiPhD** [comment]
@jni slightly off topic, but the polygon lasso issue, how are you drawing?
When i use tap/click to start and tap/click to finish, then the last click point/vertex is removed.
When I use drag (3 fingers), then when I lift off the vertex remains.
So I think this is like what we discussed here:
https://github.com/napari/napari/pull/7023#issuecomment-2185424362

Is the click-to-finish to be considered a signal to finish or an actual vertex placement?

**Comment by jni** [comment]
No, if you look at my video, several vertices can be removed at the end with lasso. I think @melonora is correct in suspecting RDP. I just find the RDP surprising when it applies to the vertices near the end of the polygon. Anyway sorry I do need to make a more detailed issue, I just haven't had a chance.

**Comment by psobolewskiPhD** [comment]
Re-opening, because there are still aspects from OP that are not fixed.

Edit: I retested and the two open issues still occur, but the rest do appear fixed. So closing since we have issues for the other two -- and PR to fix one already. Sorry for the noise.
